src/async_wrap.cc

```diff
@@ -206,23 +207,23 @@ PromiseWrap* PromiseWrap::New(Environment* env,
   obj->SetInternalField(PromiseWrap::kIsChainedPromiseField,
                         parent_wrap != nullptr ? v8::True(env->isolate())
                                                : v8::False(env->isolate()));
+  PromiseWrap* wrap = new PromiseWrap(env, obj, silent);
+  // Set a pointer to the PromiseWrap in the Promise object.
+  // The pointer is retrieved using extractPromiseWrap
   CHECK_NULL(promise->GetAlignedPointerFromInternalField(0));
-  promise->SetInternalField(0, obj);
-  return new PromiseWrap(env, obj, silent);
+  promise->SetAlignedPointerInInternalField(0, wrap);
```

Comment:
```suggestion
  promise->SetAlignedPointerInInternalField(0, wrap);
```

This has different behaviour – it breaks the connection from the Promise to the PromiseWrap JS object as visible by GC. Please keep this code as-is.

If you *do* want to make changes to address Ben’s concerns, then change the `::IsObject()` check below to a separate call to `GetAlignedPointerFromInternalField()` and compare that against `nullptr`. I don’t think either variant is 100 % supported, but V8 isn’t going to change its internal object representation overnight (or ever, probably).

Reply:
hmmm... I suspected adding the `GetAlignedPointerFromInternalField()` here would fail once the `SetInternalField()` is called because the value is not an aligned pointer :-) ... and yep... ;-)

```
[err] FATAL ERROR: v8::Object::GetAlignedPointerFromInternalField() Unaligned pointer
```

Reply:
Okay, then I’d just leave the code as it was before – I think there’s no other way to deal with this rn…